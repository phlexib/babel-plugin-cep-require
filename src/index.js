module.exports.cepRequire = function(babel) {
  const { types: t } = babel;

  return {
    name: "cep-require",
    visitor: {
      VariableDeclaration(path, state) {
        if (path.node.leadingComments) {
          if (path.node.leadingComments[0].value == "@ts-ignore-cep") {
            console.log(state.opts);
            path.node.declarations[0].init.callee.name =
              "window.cep_node.require";
          }
        }
      }
    }
  };
};
