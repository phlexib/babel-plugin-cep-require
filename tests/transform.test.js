import { transform } from "@babel/core";
import * as babel from "@babel/core";
import { cepRequire } from "../src/index";

const example = `
//@ts-ignore-cep
const fs = require("fs");
//@ts-ignore-cep
const path = require("path");
//@ts-ignore
const dateFns = require("date-fns");
`;

it("first 2 requires to be replace with window.cep_node.require", () => {
  const { code } = babel.transform(example, { plugins: [cepRequire] });
  expect(code).toMatchSnapshot();
});
