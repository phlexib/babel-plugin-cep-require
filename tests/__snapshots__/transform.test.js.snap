// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first 2 requires to be replace with window.cep_node.require 1`] = `
"//@ts-ignore-cep
const fs = window.cep_node.require(\\"fs\\"); //@ts-ignore-cep

const path = window.cep_node.require(\\"path\\"); //@ts-ignore

const dateFns = require(\\"date-fns\\");"
`;
